### PR TITLE
Odin-Weather App

### PR DESCRIPTION
**Pull request by alimJalanTech**
## Project: Odin Weather App

Live Link: https://alimjalantech.github.io/odin-WeatherApp/

## Development:

**1. Structure**

- HTML File consisting of code structure and relevant links to the external files.

**2. Styling**

- CSS File consisting of relevant styling to make the Application pleasant and accessible to the user.

**3. Scripting**

- To render the input city weather details
- To use asynchronous functions for live details 

## UI:

image

## Key Takeaways:

- API Key to access data from third party functions
- Access and Manipulating  JSON files
- Working with Async functions

## Meta Info

Reviewee: @self
Reviewer:  @balwant-jtc

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const path = require('path');
+
+module.exports = {
+  entry: './src/index.js',
+  output: {
+    filename: 'main.js',
+    path: path.resolve(__dirname, 'dist'),
+  },
+};
```
Comment (balwant-jtc) [inline webpack.config.js]: Add new line at the end

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+node_modules/*
```
Comment (balwant-jtc) [inline .gitignore:1]: add `dist` to .gitignore

File: dist/index.html [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+<body>
+    <div class="card">
+        <div class="search">
+            <input type="text" 
+            placeholder="Enter the name of City"
+            >
+            <button class="btn">Submit</button>
+        </div>
+        <div class="weather">
+            <img src="rain.png" class="Weather Icon">
```
Comment (balwant-jtc) [inline dist/index.html]: `img` tag should always contain `alt` attribute

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+const API_KEY = "aaebd4178fcf227a3349fb75ef211df7";
```
Comment (balwant-jtc) [inline src/index.js]: A good practice is to move sensitive information to separate file, such as config.js and import them here to use the API_KEY and API_URL.

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+const API_URL = "https://api.openweathermap.org/data/2.5/weather/?units=metric&q=";
+
+const searchBox = document.querySelector(".search input");
+const searchBtn = document.querySelector(".btn");
+
+async function checkWeather (city) {
+    const response = await fetch(API_URL + city+  `&appid=${API_KEY}`);
```
Comment (balwant-jtc) [inline src/index.js]: use template literal to concat url

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    var data = await response.json();
```
Comment (balwant-jtc) [inline src/index.js]: use const here instead of var

File: src/index.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+    document.querySelector(".city").innerHTML = data.name;
+    document.querySelector(".temp").innerHTML = Math.round(data.main.temp) + "°C";
```
Comment (balwant-jtc) [inline src/index.js]: use template literal for string concatenation

**Review by balwant-jtc — CHANGES_REQUESTED** [review]

File: webpack.config.js (hunk 1 above)
Comment (alimJalanTech) [reply webpack.config.js]: Fixed

File: src/index.js (hunk 7 above)
Comment (alimJalanTech) [reply src/index.js]: Fixed

File: src/index.js (hunk 6 above)
Comment (alimJalanTech) [reply src/index.js]: Fixed

File: src/index.js (hunk 5 above)
Comment (alimJalanTech) [reply src/index.js]: Fixed

File: dist/index.html (hunk 3 above)
Comment (alimJalanTech) [reply dist/index.html]: Fixed

File: src/index.js (hunk 4 above)
Comment (alimJalanTech) [reply src/index.js]: Fixed

File: .gitignore (hunk 2 above)
Comment (alimJalanTech) [reply .gitignore:1]: Fixed

**Pull request merged**